clients/lsp-clangd.el

```diff
@@ -178,6 +178,49 @@ This must be set only once after loading the clang client.")
   :risky t
   :type '(repeat string))
 
+(defun lsp-clangd-find-compile-commands-dir ()
+  "Return the full directory path (or nil) that contain a compile_commands.json.
+
+If multiple compile_commands.json files are
+found in the project directory tree - ask the user to choose."
+  ;; TODO lsp-workspace-root won't find anything at the start of the workspace
+  (let ((candidate-dirs (directory-files-recursively (lsp-workspace-root) "compile_commands.json")))
+    (pcase (length candidate-dirs)
+          ;; TODO can I lsp--warn-user-in-message-and-log at the same time
+      (`0 (lsp--warn "Cannot find compile_commands.json - intellisense might not work")
```

Comment:
using lsp--warn is a bit more for me since there are projects that may work without compile_commands. I suggest using lsp-log which won't be visible by the user.

Reply:
Is explicit UX not better? I have yet to see a big cmake-based cpp project that works out of the box.

Reply:
My concern is that a beginner opening a standalone cpp file will see the error and he/she will be confused/misled.

Reply:
The alternative is to change it to info and change the wording to something like: `Starting without compile_commands.json` without implying that there might be an error.

Reply:
> My concern is that a beginner opening a standalone cpp file will see the error and he/she will be confused/misled.

Fair point. What about something like

> Starting without compile_commands.json - some features might not work

Reply:
Thinking a bit more about that seems like it will be better to have that message visible to the user. There are a lot of users who were confused by clangd not being able to handle their project due to missing compile_commands.json. Lets have it like that:

> Starting without compile_commands.json(without this clagnd might not be able to handle complex project).

(or some alternative wording with the same structure).

Reply:
Don't forget that clangd also supports [compile_flags.txt](https://releases.llvm.org/8.0.0/tools/clang/tools/extra/docs/clangd/Installation.html#compile-flags-txt).

Reply:
> Don't forget that clangd also supports [compile_flags.txt](https://releases.llvm.org/8.0.0/tools/clang/tools/extra/docs/clangd/Installation.html#compile-flags-txt).

Thanks for the tip-off, however, it doesn't look like --compile-commands-dir will recognise compile_flags.txt.